diff -Narup dmenu-4.4.1/draw.c 00_dmenu-4.4.1_utf8widechars/draw.c
--- dmenu-4.4.1/draw.c	2011-09-19 11:48:13.000000000 +0200
+++ 00_dmenu-4.4.1_utf8widechars/draw.c	2011-10-03 17:47:38.405860362 +0200
@@ -46,7 +46,7 @@ drawtextn(DC *dc, const char *text, size
 
 	XSetForeground(dc->dpy, dc->gc, FG(dc, col));
 	if(dc->font.set)
-		XmbDrawString(dc->dpy, dc->canvas, dc->font.set, dc->gc, x, y, text, n);
+		Xutf8DrawString(dc->dpy, dc->canvas, dc->font.set, dc->gc, x, y, text, n);
 	else {
 		XSetFont(dc->dpy, dc->gc, dc->font.xfont->fid);
 		XDrawString(dc->dpy, dc->canvas, dc->gc, x, y, text, n);
@@ -165,7 +165,7 @@ textnw(DC *dc, const char *text, size_t
 	if(dc->font.set) {
 		XRectangle r;
 
-		XmbTextExtents(dc->font.set, text, len, NULL, &r);
+		Xutf8TextExtents(dc->font.set, text, len, NULL, &r);
 		return r.width;
 	}
 	return XTextWidth(dc->font.xfont, text, len);
