diff -Narup 00_dwm-4.7_bstack/config.def.h 01_dwm-4.7_maximize/config.def.h
--- 00_dwm-4.7_bstack/config.def.h	2007-11-24 14:52:16.000000000 +0100
+++ 01_dwm-4.7_maximize/config.def.h	2007-12-24 17:50:16.000000000 +0100
@@ -27,11 +27,13 @@ Rule rules[] = {
 #define RESIZEHINTS		True	/* False - respect size hints in tiled resizals */
 #define SNAP			32	/* snap pixel */
 #include "bstack.c"
+#include "maximize.c"
 Layout layouts[] = {
 	/* symbol		function */
 	{ "[]=",		tile }, /* first entry is default */
 	{ "><>",		floating },
 	{ "TTT",		bstack },
+	{ "[ ]",		maximize },
 };
 
 /* key definitions */
diff -Narup 00_dwm-4.7_bstack/maximize.c 01_dwm-4.7_maximize/maximize.c
--- 00_dwm-4.7_bstack/maximize.c	1970-01-01 01:00:00.000000000 +0100
+++ 01_dwm-4.7_maximize/maximize.c	2007-12-24 17:48:33.000000000 +0100
@@ -0,0 +1,17 @@
+void
+maximize(void) {
+	Client *c;
+        
+        for(c = clients; c; c = c->next)
+                if (isvisible(c)) {
+                        unban(c);
+                        if(! c->isfloating)
+                                resize(c, wax, way, waw - 2 * BORDERPX, wah - 2 * BORDERPX, True);
+                } else {
+                        ban(c);
+                }
+	
+        focus(NULL);
+        restack();
+}
+
