diff -Narup 01_dmenu-4.0_fixemptylines/dmenu.c 02_dmenu-4.0_utf8widechars/dmenu.c
--- 01_dmenu-4.0_fixemptylines/dmenu.c	2009-04-22 00:47:17.000000000 +0200
+++ 02_dmenu-4.0_utf8widechars/dmenu.c	2009-04-22 00:47:18.000000000 +0200
@@ -278,7 +278,7 @@ drawtext(const char *text, unsigned long
 		for(i = len; i && i > len - 3; buf[--i] = '.');
 	XSetForeground(dpy, dc.gc, col[ColFG]);
 	if(dc.font.set)
-		XmbDrawString(dpy, dc.drawable, dc.font.set, dc.gc, x, y, buf, len);
+		Xutf8DrawString(dpy, dc.drawable, dc.font.set, dc.gc, x, y, buf, len);
 	else
 		XDrawString(dpy, dc.drawable, dc.gc, x, y, buf, len);
 }
@@ -723,7 +723,7 @@ textnw(const char *text, unsigned int le
 	XRectangle r;
 
 	if(dc.font.set) {
-		XmbTextExtents(dc.font.set, text, len, NULL, &r);
+		Xutf8TextExtents(dc.font.set, text, len, NULL, &r);
 		return r.width;
 	}
 	return XTextWidth(dc.font.xfont, text, len);
