diff -Narup 01_dmenu-3.6_fixemptylines/dmenu.c 02_dmenu-3.6_utf8widechars/dmenu.c
--- 01_dmenu-3.6_fixemptylines/dmenu.c	2008-04-16 21:36:19.000000000 +0200
+++ 02_dmenu-3.6_utf8widechars/dmenu.c	2008-04-16 21:36:19.000000000 +0200
@@ -291,7 +291,7 @@ drawtext(const char *text, unsigned long
 		return; /* too long */
 	XSetForeground(dpy, dc.gc, col[ColFG]);
 	if(dc.font.set)
-		XmbDrawString(dpy, dc.drawable, dc.font.set, dc.gc, x, y, buf, len);
+		Xutf8DrawString(dpy, dc.drawable, dc.font.set, dc.gc, x, y, buf, len);
 	else
 		XDrawString(dpy, dc.drawable, dc.gc, x, y, buf, len);
 }
@@ -733,7 +733,7 @@ textnw(const char *text, unsigned int le
 	XRectangle r;
 
 	if(dc.font.set) {
-		XmbTextExtents(dc.font.set, text, len, NULL, &r);
+		Xutf8TextExtents(dc.font.set, text, len, NULL, &r);
 		return r.width;
 	}
 	return XTextWidth(dc.font.xfont, text, len);
