diff -Narup 02_dmenu-3.9_utf8widechars/config.mk 03_dmenu-3.9_patchversion/config.mk
--- 02_dmenu-3.9_utf8widechars/config.mk	2008-09-20 23:26:42.000000000 +0200
+++ 03_dmenu-3.9_patchversion/config.mk	2008-09-20 23:26:42.000000000 +0200
@@ -3,6 +3,9 @@ VERSION = 3.9
 
 # Customize below to fit your system
 
+# dwm-mitch patchlevel
+VERSION := ${VERSION}+${PATCHVERSION}
+
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
