diff -Narup 03_dwm-4.7_perworkspace/dwm.c 04_dwm-4.7_clientstyle/dwm.c
--- 03_dwm-4.7_perworkspace/dwm.c	2007-12-25 13:24:48.000000000 +0100
+++ 04_dwm-4.7_clientstyle/dwm.c	2007-12-25 13:27:35.000000000 +0100
@@ -130,7 +130,6 @@ void detach(Client *c);
 void detachstack(Client *c);
 void detachhiddenstack(Client *c);
 void drawbar(void);
-void drawsquare(Bool filled, Bool empty, unsigned long col[ColLast]);
 void drawtext(const char *text, unsigned long col[ColLast]);
 void enternotify(XEvent *e);
 void eprint(const char *errstr, ...);
@@ -563,7 +562,6 @@ drawbar(void) {
 		dc.x = x;
 		if(sel) {
 			drawtext(sel->name, dc.sel);
-			drawsquare(sel->ismax, sel->isfloating, dc.sel);
 		}
 		else
 			drawtext(NULL, dc.norm);
@@ -573,27 +571,6 @@ drawbar(void) {
 }
 
 void
-drawsquare(Bool filled, Bool empty, unsigned long col[ColLast]) {
-	int x;
-	XGCValues gcv;
-	XRectangle r = { dc.x, dc.y, dc.w, dc.h };
-
-	gcv.foreground = col[ColFG];
-	XChangeGC(dpy, dc.gc, GCForeground, &gcv);
-	x = (dc.font.ascent + dc.font.descent + 2) / 4;
-	r.x = dc.x + 1;
-	r.y = dc.y + 1;
-	if(filled) {
-		r.width = r.height = x + 1;
-		XFillRectangles(dpy, dc.drawable, dc.gc, &r, 1);
-	}
-	else if(empty) {
-		r.width = r.height = x;
-		XDrawRectangles(dpy, dc.drawable, dc.gc, &r, 1);
-	}
-}
-
-void
 drawtext(const char *text, unsigned long col[ColLast]) {
 	int x, y, w, h;
 	static char buf[256];
