diff -Narup 05_dwm-4.7_utf8widechars/config.mk 06_dwm-4.7_patchversion/config.mk
--- 05_dwm-4.7_utf8widechars/config.mk	2010-02-22 19:57:07.000000000 +0100
+++ 06_dwm-4.7_patchversion/config.mk	2010-02-22 19:57:07.000000000 +0100
@@ -3,6 +3,9 @@ VERSION = 4.7
 
 # Customize below to fit your system
 
+# dwm-mitch patchlevel
+VERSION := ${VERSION}+${PATCHVERSION}
+
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
