diff -Narup 08_dwm-4.7_floatborder/dwm.c 09_dwm-4.7_maxfloat/dwm.c
--- 08_dwm-4.7_floatborder/dwm.c	2007-12-26 18:40:37.000000000 +0100
+++ 09_dwm-4.7_maxfloat/dwm.c	2007-12-26 18:43:16.000000000 +0100
@@ -67,7 +67,7 @@ struct Client {
 	int minax, maxax, minay, maxay;
 	long flags;
 	unsigned int border, oldborder, workspace;
-	Bool isbanned, isfixed, ismax, isfloating, wasfloating;
+	Bool isbanned, isfixed, ismax, isfloating;
 	Client *next;
 	Client *prev;
 	Client *snext;
@@ -1642,13 +1642,9 @@ togglemax(const char *arg) {
 
 	if(!sel || sel->isfixed)
 		return;
+	if((layout[selws-1]->arrange != floating) && ! sel->isfloating))
+		return;
 	if((sel->ismax = !sel->ismax)) {
-		if((layout[selws-1]->arrange == floating) || sel->isfloating)
-			sel->wasfloating = True;
-		else {
-			togglefloating(NULL);
-			sel->wasfloating = False;
-		}
 		sel->rx = sel->x;
 		sel->ry = sel->y;
 		sel->rw = sel->w;
@@ -1657,10 +1653,7 @@ togglemax(const char *arg) {
 	}
 	else {
 		resize(sel, sel->rx, sel->ry, sel->rw, sel->rh, True);
-		if(!sel->wasfloating)
-			togglefloating(NULL);
 	}
-	drawbar();
 	while(XCheckMaskEvent(dpy, EnterWindowMask, &ev));
 }
 
