diff -Narup dmenu-3.4/config.h 99_dmenu-3.4_personal_configuration.mitch/config.h
--- dmenu-3.4/config.h	2007-10-28 19:43:55.000000000 +0100
+++ 99_dmenu-3.4_personal_configuration.mitch/config.h	2007-10-28 19:44:30.000000000 +0100
@@ -1,11 +1,11 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-#define FONT			"-*-terminus-medium-*-*-*-*-*-*-*-*-*-*-*"
+#define FONT			"-misc-fixed-medium-r-*-*-13-*-*-*-*-*-*-*"
 #define NORMBGCOLOR             "#cccccc"
 #define NORMFGCOLOR             "#000000"
 #define SELBGCOLOR              "#0066ff"
 #define SELFGCOLOR              "#ffffff"
 /* next macro defines the space between menu items */
 #define SPACE			30 /* px */
-#define HIST_SIZE       	20
+#define HIST_SIZE       	35
diff -Narup dmenu-3.4/config.h.orig 99_dmenu-3.4_personal_configuration.mitch/config.h.orig
--- dmenu-3.4/config.h.orig	1970-01-01 01:00:00.000000000 +0100
+++ 99_dmenu-3.4_personal_configuration.mitch/config.h.orig	2007-10-28 19:43:55.000000000 +0100
@@ -0,0 +1,11 @@
+/* See LICENSE file for copyright and license details. */
+
+/* appearance */
+#define FONT			"-*-terminus-medium-*-*-*-*-*-*-*-*-*-*-*"
+#define NORMBGCOLOR             "#cccccc"
+#define NORMFGCOLOR             "#000000"
+#define SELBGCOLOR              "#0066ff"
+#define SELFGCOLOR              "#ffffff"
+/* next macro defines the space between menu items */
+#define SPACE			30 /* px */
+#define HIST_SIZE       	20
diff -Narup dmenu-3.4/config.mk 99_dmenu-3.4_personal_configuration.mitch/config.mk
--- dmenu-3.4/config.mk	2007-10-28 19:43:55.000000000 +0100
+++ 99_dmenu-3.4_personal_configuration.mitch/config.mk	2007-10-28 19:43:55.000000000 +0100
@@ -4,7 +4,7 @@ VERSION = 3.4
 # Customize below to fit your system
 
 # dwm-mitch patchlevel
-VERSION := ${VERSION}+${PATCHVERSION}
+VERSION := ${VERSION}+${PATCHVERSION}+pm
 
 # paths
 PREFIX = /usr/local
