diff -Narup dmenu-3.6/config.h 99_dmenu-3.6_personal_configuration.mitch/config.h
--- dmenu-3.6/config.h	2008-04-16 21:39:06.000000000 +0200
+++ 99_dmenu-3.6_personal_configuration.mitch/config.h	2008-04-16 21:39:40.000000000 +0200
@@ -1,11 +1,11 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-#define FONT			"-*-terminus-medium-r-normal-*-14-*-*-*-*-*-*-*"
+#define FONT			"-misc-fixed-medium-r-*-*-13-*-*-*-*-*-*-*"
 #define NORMBGCOLOR             "#cccccc"
 #define NORMFGCOLOR             "#000000"
 #define SELBGCOLOR              "#0066ff"
 #define SELFGCOLOR              "#ffffff"
 /* next macro defines the space between menu items */
 #define SPACE			30 /* px */
-#define HIST_SIZE       	20
+#define HIST_SIZE       	35
