diff -Narup dmenu-3.8/config.h 99_dmenu-3.8_personal_configuration.mitch/config.h
--- dmenu-3.8/config.h	2008-07-31 20:16:33.000000000 +0200
+++ 99_dmenu-3.8_personal_configuration.mitch/config.h	2008-07-31 20:16:33.000000000 +0200
@@ -1,11 +1,11 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const char *font        = "-*-terminus-medium-r-normal-*-14-*-*-*-*-*-*-*";
+static const char *font        = "-misc-fixed-medium-r-normal-*-13-*-*-*-*-*-*-*";
 static const char *normbgcolor = "#cccccc";
 static const char *normfgcolor = "#000000";
 static const char *selbgcolor  = "#0066ff";
 static const char *selfgcolor  = "#ffffff";
 static unsigned int spaceitem  = 30; /* px between menu items */
 
-#define HIST_SIZE              20
+#define HIST_SIZE              35
