diff -Narup dwm-4.7/config.def.h 99_dwm-4.7_personal_configuration.andy/config.def.h
--- dwm-4.7/config.def.h	2009-01-29 17:41:14.527099961 +0100
+++ 99_dwm-4.7_personal_configuration.andy/config.def.h	2009-01-29 17:48:02.957549752 +0100
@@ -4,13 +4,13 @@
 #define BARPOS			BarTop /* BarBot, BarOff */
 #define BORDERPX		1
 #define FLOATBORDERPX		1
-#define FONT			"-*-terminus-medium-*-*-*-*-*-*-*-*-*-*-*"
-#define NORMBORDERCOLOR		"#cccccc"
-#define NORMBGCOLOR		"#cccccc"
-#define NORMFGCOLOR		"#000000"
-#define SELBORDERCOLOR		"#0066ff"
-#define SELBGCOLOR		"#0066ff"
-#define SELFGCOLOR		"#ffffff"
+#define FONT			"-misc-fixed-medium-r-normal-*-13-*-*-*-*-*-*-*"
+#define NORMBORDERCOLOR		"#080"
+#define NORMBGCOLOR		"#222"
+#define NORMFGCOLOR		"#0c0"
+#define SELBORDERCOLOR		"#9cf"
+#define SELBGCOLOR		"#555"
+#define SELFGCOLOR		"#080"
 
 /* workspaces */
 /* Query class:instance:title for regex matching info with following command:
@@ -21,6 +21,9 @@ Rule rules[] = {
 	{ "Gimp",			True,		0 },
 	{ "MPlayer",			True,		0 },
 	{ "Acroread",			True,		0 },
+	{ "[Aa]udacious",		True,		0 },
+	{ "gkrellm",			True,		0 },
+	{ "pidgin",			True,		0 },	
 };
 #define INITIALWORKSPACES	 1
 #define MAXWORKSPACES		99
@@ -33,66 +36,47 @@ Rule rules[] = {
 #include "bstack.c"
 #include "maximize.c"
 Layout layouts[] = {
-	/* symbol		function */
-	{ "[]=",		tile }, /* first entry is default */
-	{ "><>",		floating },
+	/* symbol		function || first entry is default */
 	{ "TTT",		bstack },
+	{ "[]=",		tile },
 	{ "[ ]",		maximize },
+	{ "<><",		floating }, 
 };
 
 /* key definitions */
 #define MODKEY			Mod1Mask
+/*#define SUPERKEY			Sup1Mask*/
 Key keys[] = {
 	/* modifier			key		function	argument */
-	{ MODKEY,			XK_p,		spawn,
-		"exe=`dmenu_path | dmenu -fn '"FONT"' -nb '"NORMBGCOLOR"' -nf '"NORMFGCOLOR"'"
-		" -sb '"SELBGCOLOR"' -sf '"SELFGCOLOR"'` && exec $exe" },
-	{ MODKEY|ShiftMask,		XK_Return,	spawn, "exec uxterm" },
-	{ MODKEY,			XK_space,	setlayout,	NULL },
+	{ MODKEY,			XK_t,		spawn,		"exec x-terminal-emulator" },
+	{ MODKEY|ShiftMask,		XK_y,		spawn,		"exec dwm-choose" },
+	{ MODKEY,			XK_Tab,		focusnext,	NULL },
+	{ MODKEY|ShiftMask,		XK_Tab,		focusprev,	NULL },
+	{ MODKEY,			XK_Return,	zoom,		NULL },
 	{ MODKEY,			XK_b,		togglebar,	NULL },
 	{ MODKEY,			XK_j,		focusnext,	NULL },
 	{ MODKEY,			XK_k,		focusprev,	NULL },
-	{ MODKEY,			XK_g,		setmwfact,	"-0.05" },
-	{ MODKEY,			XK_s,		setmwfact,	"+0.05" },
-/*	{ MODKEY|ShiftMask,		XK_g,		incnmaster,	"1" }, */
-/*	{ MODKEY|ShiftMask,		XK_s,		incnmaster,	"-1" }, */
-	{ MODKEY,			XK_a,		popstack,	NULL },
-	{ MODKEY,			XK_d,		pushstack,	NULL },
+	{ MODKEY|ControlMask,		XK_e,		setmwfact,	"+0.05" },
+	{ MODKEY|ControlMask,		XK_d,		setmwfact,	"-0.05" },
+/*	{ MODKEY|ShiftMask,		XK_e,		incnmaster,	"1" }, */
+/*	{ MODKEY|ShiftMask,		XK_d,		incnmaster,	"-1" }, */
+	{ MODKEY|ShiftMask,		XK_a,		popstack,	NULL },
+	{ MODKEY|ShiftMask,		XK_d,		pushstack,	NULL },
 	{ MODKEY|ControlMask,		XK_y,		wscount,	"1" },
-	{ MODKEY|ControlMask,		XK_r,		wscount,	"-1" },
-	{ MODKEY,			XK_m,		togglemax,	NULL },
-	{ MODKEY,			XK_Return,	zoom,		NULL },
-	{ MODKEY,			XK_Tab,		focusnext,	NULL },
-	{ MODKEY|ShiftMask,		XK_Tab,		focusprev,	NULL },
-	{ MODKEY|ShiftMask,		XK_space,	togglefloating,	NULL },
-	{ MODKEY|ShiftMask,		XK_c,		killclient,	NULL },
-	{ MODKEY,			XK_1,		view,		"1" },
-	{ MODKEY,			XK_2,		view,		"2" },
-	{ MODKEY,			XK_3,		view,		"3" },
-	{ MODKEY,			XK_4,		view,		"4" },
-	{ MODKEY,			XK_5,		view,		"5" },
-	{ MODKEY,			XK_6,		view,		"6" },
-	{ MODKEY,			XK_7,		view,		"7" },
-	{ MODKEY,			XK_8,		view,		"8" },
-	{ MODKEY,			XK_9,		view,		"9" },
-	{ MODKEY,			XK_0,		view,		"10" },
-	{ MODKEY|ShiftMask,		XK_1,		moveto,		"1" },
-	{ MODKEY|ShiftMask,		XK_2,		moveto,		"2" },
-	{ MODKEY|ShiftMask,		XK_3,		moveto,		"3" },
-	{ MODKEY|ShiftMask,		XK_4,		moveto,		"4" },
-	{ MODKEY|ShiftMask,		XK_5,		moveto,		"5" },
-	{ MODKEY|ShiftMask,		XK_6,		moveto,		"6" },
-	{ MODKEY|ShiftMask,		XK_7,		moveto,		"7" },
-	{ MODKEY|ShiftMask,		XK_8,		moveto,		"8" },
-	{ MODKEY|ShiftMask,		XK_9,		moveto,		"9" },
-	{ MODKEY|ShiftMask,		XK_0,		moveto,		"10" },
-	{ MODKEY,			XK_l,		viewrel,	"1" },
-	{ MODKEY,			XK_h,		viewrel,	"-1" },
-	{ MODKEY|ShiftMask,		XK_l,		pushstack,	NULL },
-	{ MODKEY|ShiftMask,		XK_l,		viewrel,	"1" },
-	{ MODKEY|ShiftMask,		XK_l,		popstack,	NULL },
-	{ MODKEY|ShiftMask,		XK_h,		pushstack,	NULL },
-	{ MODKEY|ShiftMask,		XK_h,		viewrel,	"-1" },
-	{ MODKEY|ShiftMask,		XK_h,		popstack,	NULL },
-	{ MODKEY|ShiftMask,		XK_q,		quit,		NULL },
+	{ MODKEY|ControlMask,		XK_x,		wscount,	"-1" },
+	{ MODKEY|ControlMask,		XK_c,		killclient,	NULL },
+	{ MODKEY|ShiftMask,		XK_f,		togglefloating,	NULL },
+	{ MODKEY,			XK_f,		viewrel,	"1" },
+	{ MODKEY,			XK_s,		viewrel,	"-1" },
+	{ MODKEY|ShiftMask|ControlMask,	XK_q,		quit,		NULL },
+	{ MODKEY|ShiftMask,		XK_1,		setlayout,	"<><" },
+	{ MODKEY|ShiftMask,		XK_2,		setlayout,	"[ ]" },
+	{ MODKEY|ShiftMask,		XK_3,		setlayout,	"TTT" },
+	{ MODKEY|ShiftMask,		XK_4,		setlayout,	"[]=" },
+	{ MODKEY|ShiftMask,		XK_5,		setlayout,	"(@)" },
+	{ MODKEY|ShiftMask,		XK_6,		setlayout,	"[\\]" },
+	/*	defining some global public privat shortcuts	*/
+	{ MODKEY,			XK_Up,		spawn, 		"exec aumix -w +2"},
+	{ MODKEY,			XK_Down,	spawn, 		"exec aumix -w -2"},
+	{ MODKEY|ControlMask,		XK_KP_Enter,	spawn, 		"exec xlock -mode blank"},
 };
diff -Narup dwm-4.7/config.def.h~ 99_dwm-4.7_personal_configuration.andy/config.def.h~
--- dwm-4.7/config.def.h~	1970-01-01 01:00:00.000000000 +0100
+++ 99_dwm-4.7_personal_configuration.andy/config.def.h~	2009-01-29 17:41:14.000000000 +0100
@@ -0,0 +1,80 @@
+/* See LICENSE file for copyright and license details. */
+
+/* appearance */
+#define BARPOS			BarTop /* BarBot, BarOff */
+#define BORDERPX		0
+#define FLOATBORDERPX		0
+#define FONT			"-misc-fixed-medium-r-normal-*-13-*-*-*-*-*-*-*"
+#define NORMBORDERCOLOR		"#080"
+#define NORMBGCOLOR		"#222"
+#define NORMFGCOLOR		"#0c0"
+#define SELBORDERCOLOR		"#9cf"
+#define SELBGCOLOR		"#555"
+#define SELFGCOLOR		"#080"
+
+/* workspaces */
+/* Query class:instance:title for regex matching info with following command:
+ * xprop | awk -F '"' '/^WM_CLASS/ { printf("%s:%s:",$4,$2) }; /^WM_NAME/ { printf("%s\n",$2) }' */
+Rule rules[] = {
+	/* class:instance:title regex	isfloating	workspace (0=current)*/
+	{ "Firefox",			False,		0 },
+	{ "Gimp",			True,		0 },
+	{ "MPlayer",			True,		0 },
+	{ "Acroread",			True,		0 },
+	{ "[lL]opster",			True,		0 },
+};
+#define INITIALWORKSPACES	 1
+#define MAXWORKSPACES		99
+#define MAXWSTEXTWIDTH		 6	/* must be 2*(strlen(MAXWORKSPACES)+1)  */
+
+/* layout(s) */
+#define MWFACT			0.6	/* master width factor [0.1 .. 0.9] */
+#define RESIZEHINTS		True	/* False - respect size hints in tiled resizals */
+#define SNAP			32	/* snap pixel */
+#include "bstack.c"
+#include "maximize.c"
+Layout layouts[] = {
+	/* symbol		function */
+	{ "<><",		floating }, /* first entry is default */
+	{ "[]=",		tile },
+	{ "TTT",		bstack },
+	{ "[ ]",		maximize },
+};
+
+/* key definitions */
+#define MODKEY			Mod1Mask
+/*#define SUPERKEY			Sup1Mask*/
+Key keys[] = {
+	/* modifier			key		function	argument */
+	{ MODKEY,			XK_t,		spawn,		"exec x-terminal-emulator" }, \
+	{ MODKEY|ShiftMask,		XK_y,		spawn,		"exec dwm-choose" }, \
+	{ MODKEY,			XK_Tab,		focusnext,	NULL }, \
+	{ MODKEY|ShiftMask,		XK_Tab,		focusprev,	NULL }, \
+	{ MODKEY,			XK_Return,	zoom,		NULL }, \
+	{ MODKEY,			XK_b,		togglebar,	NULL },
+	{ MODKEY,			XK_j,		focusnext,	NULL },
+	{ MODKEY,			XK_k,		focusprev,	NULL },
+	{ MODKEY|ControlMask,		XK_e,		setmwfact,	"+0.05" },
+	{ MODKEY|ControlMask,		XK_d,		setmwfact,	"-0.05" },
+/*	{ MODKEY|ShiftMask,		XK_e,		incnmaster,	"1" }, */
+/*	{ MODKEY|ShiftMask,		XK_d,		incnmaster,	"-1" }, */
+	{ MODKEY|ShiftMask,		XK_a,		popstack,	NULL },
+	{ MODKEY|ShiftMask,		XK_d,		pushstack,	NULL },
+	{ MODKEY|ControlMask,		XK_y,		wscount,	"1" },
+	{ MODKEY|ControlMask,		XK_x,		wscount,	"-1" },
+	{ MODKEY|ControlMask,		XK_c,		killclient,	NULL },
+	{ MODKEY|ShiftMask,		XK_f,		togglefloating,	NULL },
+	{ MODKEY,			XK_f,		viewrel,	"1" },
+	{ MODKEY,			XK_s,		viewrel,	"-1" },
+	{ MODKEY|ShiftMask|ControlMask,	XK_q,		quit,		NULL },
+	{ MODKEY|ShiftMask,		XK_1,		setlayout,	"<><" },
+	{ MODKEY|ShiftMask,		XK_2,		setlayout,	"[ ]" },
+	{ MODKEY|ShiftMask,		XK_3,		setlayout,	"TTT" },
+	{ MODKEY|ShiftMask,		XK_4,		setlayout,	"[]=" },
+	{ MODKEY|ShiftMask,		XK_5,		setlayout,	"(@)" },
+	{ MODKEY|ShiftMask,		XK_6,		setlayout,	"[\\]" },
+	/*	defining some global privat shortcuts	*/
+	{ MODKEY,			XK_Up,		spawn, 		"exec aumix -w +2"},
+	{ MODKEY,			XK_Down,	spawn, 		"exec aumix -w -2"},
+	{ MODKEY|ControlMask,		XK_KP_Enter,	spawn, 		"exec xlock -mode blank"},
+};
diff -Narup dwm-4.7/config.mk 99_dwm-4.7_personal_configuration.andy/config.mk
--- dwm-4.7/config.mk	2009-01-29 17:41:14.517102797 +0100
+++ 99_dwm-4.7_personal_configuration.andy/config.mk	2009-01-29 17:41:14.589093251 +0100
@@ -4,7 +4,7 @@ VERSION = 4.7
 # Customize below to fit your system
 
 # dwm-mitch patchlevel
-VERSION := ${VERSION}+${PATCHVERSION}
+VERSION := ${VERSION}+${PATCHVERSION}+pa
 
 # paths
 PREFIX = /usr/local
